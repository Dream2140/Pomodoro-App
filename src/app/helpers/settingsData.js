import { firebaseDB } from '../firebase-api';
import { eventBus } from './eventBus';

/**
 * @description manages db and session storage connections
 * @exports SettingsData
 * @class SettingsData
 */
class SettingsData {

  /**
   * @description Creates an instance of SettingsData
   * @param {object} base instance of Firebase
   * @memberof SettingsData
   */
  constructor(base) {
    this.db = base;
    eventBus.subscribe('settings-page-loading', () =>
      this.receiveData('settings')
    );

  }

  /**
   * @description receives data from db
   * @param {string} key
   * @return {object} result
   * @memberof SettingsData
   */
  async receiveData(key) {
    try {
      const result = await this.db.getData(key);
      this.setDataToStorage(key, this.db.data);
      eventBus.post('load-page');
      return result;
    } catch (error) {
      console.error(error);
    }
  }

  /**
   * @description sends data to firebase
   * @param {string} key
   * @param {object} data
   * @param {string} id
   * @return {boolean}
   * @memberof SettingsData
   */
  async sendData(key, data, id) {
    return await this.db.sendData(key, data, id);
  }

  async registerUser(email,password) {
    return await this.db.registerUser(email,password);
  }

  async loginUser(email,password) {
    return await this.db.loginUser(email,password);
  }

  /**
   * @description sets data to session storage
   * @param {string} key
   * @param {*} value
   * @memberof SettingsData
   */
  setDataToStorage(key, value) {
    sessionStorage.setItem(key, JSON.stringify(value));
  }

  /**
   * @description receives data from session storage
   * @param {string} key
   * @return {*} data from storage
   * @memberof SettingsData
   */
  getDataFromStorage(key) {
    return JSON.parse(sessionStorage.getItem(key));
  }

  /**
   * @description removes items from firebase
   * @param {string} key
   * @param {object} ids ids of tasks' to remove
   * @return {object} promise
   * @memberof SettingsData
   */
  removeItem(key, ids) {
    return this.db.removeItem(key, ids);
  }
}

export const settingsData = new SettingsData(firebaseDB);