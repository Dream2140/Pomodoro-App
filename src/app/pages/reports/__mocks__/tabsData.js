export const tabsData = {
    leftTabs: false,
    typeTabs: 'reports__top-tabs',
    tabPosition: 'tab__item--centered',
    tabData: [

      {
        period: true,
        val: 'day',
        data: 'Day',
        dataTab: 'data-reports-period="day"'
      },
      {
        period: true,
        val: 'week',
        data: 'Week',
        dataTab: 'data-reports-period="week"'
      },
      {
        period: true,
        val: 'month',
        data: 'Month',
        dataTab: 'data-reports-period="month"'
      }
    ]
};
