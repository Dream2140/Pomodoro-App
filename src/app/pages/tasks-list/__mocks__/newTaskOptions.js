export const newTaskOptions = {
  categoryId: 'other',
  completeDate: null,
  completedCount: [],

  deadline: 'April 5, 2022',
  
  description: 'Hello! Let`s start cleaning!',
  estimation: '5',
  failedPomodoros: [],
  priority: 'middle',
  status: 'GLOBAL_LIST',
  title: 'Test',
};
