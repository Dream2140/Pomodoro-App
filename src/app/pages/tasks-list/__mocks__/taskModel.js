export class TaskModel {
  constructor(options) {
    this.options = {
      title : options.title,
      description : options.description,
      failedPomodoros : options.failedPomodoros,
      completeDate : options.completeDate,
      completedCount : options.completedCount,
      priority : options.priority,
      categoryId : options.categoryId,
      status : options.status,
      estimation : options.estimation,
      deadline : options.deadline,
    }
  }
}