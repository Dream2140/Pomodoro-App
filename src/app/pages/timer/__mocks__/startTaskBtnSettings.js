export const startTaskBtnSettings = {
    secondBtn: true,
    firstColor: 'red',
    firstServiceClass: 'fail-pomodora',
    firstData: 'Fail Pomodora',
    secondColor: 'green',
    secondServiceClass: 'finish-pomodora',
    secondData: 'Finish Pomodora'
};