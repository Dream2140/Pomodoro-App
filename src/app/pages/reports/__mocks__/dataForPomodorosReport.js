import { Helpers } from '../../../helpers/helpers';

export const dataForPomodorosReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 7,
    high: 1,
    low: 0,
    middle: 0,
    urgent: 3,
  },
];
