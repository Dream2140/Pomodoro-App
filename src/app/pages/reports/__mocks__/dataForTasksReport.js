import { Helpers } from '../../../helpers/helpers';

export const dataForTasksReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 1,
    high: 0,
    low: 0,
    middle: 0,
    urgent: 1,
  },
];
