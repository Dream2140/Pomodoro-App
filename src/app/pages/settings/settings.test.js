/* eslint-disable no-undef */
import { settingsData } from './__mocks__/settingsData';
import { buttonsValues } from './__mocks__/buttonsValues';
import settingsTemplate from './template/settings.hbs';
import categoriesTemplate from './template/settingsCategories.hbs';

import { SettingsModel } from './scripts/settingsModel';
import { allSettingsData } from './__mocks__/allSettingsData';
import { fullCycleTime } from './__mocks__/fullCycleTime';
import { firstCycleTime } from './__mocks__/firstCycleTime';

const settingsModel = new SettingsModel();

describe('rendering', () => {
  it('renders correctly page', () => {
    const tree = JSON.stringify(settingsTemplate(settingsData));
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly categories', () => {
    const tree = JSON.stringify(
      categoriesTemplate({
        tab: {
          firstVal: 'pomodoros',
          secondVal: 'categories',
          activeTab: true,
          firstValAttr: 'data-tab="pomodoros"',
          secondValAttr: 'data-tab="categories"',
        },
      })
    );
    expect(tree).toMatchSnapshot();
  });
});


describe('getting settings data', () => {
  beforeEach(() => {
    settingsModel.buttonsValues = buttonsValues;
  });
  it('should return settings data', () => {
    const result = settingsModel.getAllSettingsData();
    expect(result).toEqual(allSettingsData);
  });

  it('should return full cycle time data', () => {
    const result = settingsModel.getFullCycleTime();
    expect(result).toEqual(fullCycleTime);
  });

  it('should return first cycle time data', () => {
    const result = settingsModel.getFirstCycleTime();
    expect(result).toEqual(firstCycleTime);
  });
});

describe('increasing/decreasing values', () => {
  beforeEach(() => {
    settingsModel.buttonsValues = buttonsValues;
    jest.spyOn(settingsModel, 'setValue');
  });

  it('should call set button value func', () => {
    settingsModel.increaseValue(['longBreak', 20]);
    expect(settingsModel.setValue).toBeCalled();
    expect(settingsModel.setValue).toHaveBeenCalledTimes(1);
  });

  it('should call set button value func', () => {
    settingsModel.decreaseValue(['shortBreak', 5]);
    expect(settingsModel.setValue).toBeCalled();
    expect(settingsModel.setValue).toHaveBeenCalledTimes(1);
  });

  it('should not call set button value func', () => {
    const result = settingsModel.decreaseValue([3456, 5]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  it('should not call set button value func', () => {
    const result = settingsModel.decreaseValue([true, false]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  it('should not call set button value func', () => {
    const result = settingsModel.decreaseValue();
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  it('should not call set button value func', () => {
    const result = settingsModel.increaseValue([3456, 5]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  it('should not call set button value func', () => {
    const result = settingsModel.increaseValue([true, false]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  it('should not call set button value func', () => {
    const result = settingsModel.increaseValue();
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });
});

describe('setting values', () => {
  beforeEach(() => {
    settingsModel.buttonsValues = buttonsValues;
  });

  it('should set button value', () => {
    const arg = ['work', 25];
    const [id, currentValue] = arg;
    const newValue = currentValue - settingsModel.buttonsValues[id].step;

    settingsModel.setValue(id, newValue);
    expect(settingsModel.buttonsValues[id].value).toBe(newValue);
  });

  it('should not set button value', () => {
    settingsModel.setValue('work', 45);
    expect(settingsModel.buttonsValues['work'].value).toBe(20);
  });

  it('should return falsy value', () => {
    const res = settingsModel.setValue(34, 56);
    expect(res).toBeFalsy();
  });

  it('should return falsy value', () => {
    const res = settingsModel.setValue();
    expect(res).toBeFalsy();
  });
});
