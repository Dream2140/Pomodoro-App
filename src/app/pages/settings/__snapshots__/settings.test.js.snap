// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly categories 1`] = `"\\"<header class=\\\\\\"header \\\\\\">\\\\r\\\\n    <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n        <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                    <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                </a>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                            data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                </ul>\\\\r\\\\n            </nav>\\\\r\\\\n\\\\r\\\\n        </div>\\\\r\\\\n\\\\r\\\\n    </div>\\\\r\\\\n</header><main class=\\\\\\"content settings-page\\\\\\">\\\\r\\\\n\\\\r\\\\n    <section class=\\\\\\"settings\\\\\\">\\\\r\\\\n        <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n            <div class=\\\\\\"tab \\\\\\">\\\\r\\\\n                <div class=\\\\\\"tab__container\\\\\\">\\\\r\\\\n                    <ul class=\\\\\\"tab__item \\\\\\">\\\\r\\\\n                    </ul>\\\\r\\\\n                </div>\\\\r\\\\n            </div>            <div class=\\\\\\"settings__container\\\\\\">\\\\r\\\\n                <h2 class=\\\\\\"settings__title\\\\\\">Categories list overview</h2>\\\\r\\\\n                <ul class=\\\\\\"settings__options setting-list\\\\\\">\\\\r\\\\n                    <li class=\\\\\\"setting-list__item setting-list__item--work\\\\\\">\\\\r\\\\n                        Work\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"setting-list__item setting-list__item--education\\\\\\">\\\\r\\\\n                        Education\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"setting-list__item setting-list__item--other\\\\\\">\\\\r\\\\n                        Other\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"setting-list__item setting-list__item--sport\\\\\\">\\\\r\\\\n                        Sport\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"setting-list__item setting-list__item--hobby\\\\\\">\\\\r\\\\n                        Hobby\\\\r\\\\n                    </li>\\\\r\\\\n                </ul>\\\\r\\\\n\\\\r\\\\n                <div class=\\\\\\"settings__buttons setting-list__btn service-btn service-btn--single\\\\\\">\\\\r\\\\n                    <button class=\\\\\\"service-btn__btn service-btn__btn--blue settings__buttons-skip\\\\\\">Go to Tasks</button>\\\\r\\\\n                </div>\\\\r\\\\n            </div>\\\\r\\\\n        </div>\\\\r\\\\n    </section>\\\\r\\\\n\\\\r\\\\n</main>\\""`;

exports[`rendering renders correctly page 1`] = `"\\"\\\\r\\\\n    <header class=\\\\\\"header \\\\\\">\\\\r\\\\n        <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n            <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n    \\\\r\\\\n                <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                    <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                        <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                    </a>\\\\r\\\\n                </div>\\\\r\\\\n    \\\\r\\\\n                <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                    <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n                </div>\\\\r\\\\n    \\\\r\\\\n                <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                        <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                            <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                                <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                            </button>\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                            <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                                <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                            </button>\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                            <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                                data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                                <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                            </button>\\\\r\\\\n                        </li>\\\\r\\\\n                    </ul>\\\\r\\\\n                </nav>\\\\r\\\\n    \\\\r\\\\n            </div>\\\\r\\\\n    \\\\r\\\\n        </div>\\\\r\\\\n    </header>    <main class=\\\\\\"content settings-page\\\\\\">\\\\r\\\\n\\\\r\\\\n        <section class=\\\\\\"settings\\\\\\">\\\\r\\\\n            <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n                <div class=\\\\\\"tab \\\\\\">\\\\r\\\\n                    <div class=\\\\\\"tab__container\\\\\\">\\\\r\\\\n                        <ul class=\\\\\\"tab__item \\\\\\">\\\\r\\\\n                        </ul>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>                <div class=\\\\\\"settings__container d-none\\\\\\">\\\\r\\\\n                    <h2 class=\\\\\\"settings__title\\\\\\">Categories list overview</h2>\\\\r\\\\n                    <ul class=\\\\\\"settings__options setting-list\\\\\\">\\\\r\\\\n                        <li class=\\\\\\"setting-list__item setting-list__item--work\\\\\\">\\\\r\\\\n                            Work\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"setting-list__item setting-list__item--education\\\\\\">\\\\r\\\\n                            Education\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"setting-list__item setting-list__item--other\\\\\\">\\\\r\\\\n                            Other\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"setting-list__item setting-list__item--sport\\\\\\">\\\\r\\\\n                            Sport\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"setting-list__item setting-list__item--hobby\\\\\\">\\\\r\\\\n                            Hobby\\\\r\\\\n                        </li>\\\\r\\\\n                    </ul>\\\\r\\\\n\\\\r\\\\n                    <div class=\\\\\\"settings__buttons setting-list__btn service-btn service-btn--single\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"service-btn__btn service-btn__btn--blue\\\\\\">Skip</button>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>\\\\r\\\\n                <!-- remove d-none -->\\\\r\\\\n                <div class=\\\\\\"settings__container\\\\\\">\\\\r\\\\n                    <h2 class=\\\\\\"settings__title\\\\\\">Pomodoros settings</h2>\\\\r\\\\n                    <ul class=\\\\\\"settings__change change-list\\\\\\">\\\\r\\\\n                        <li class=\\\\\\"change-list__item change-list__item--work\\\\\\" id=\\\\\\"\\\\\\">\\\\r\\\\n                            <div class=\\\\\\"change-list__info\\\\\\">\\\\r\\\\n                                <h4 class=\\\\\\"change-list__title\\\\\\">Work time</h4>\\\\r\\\\n                                <div class=\\\\\\"change-list__setting\\\\\\" id=\\\\\\"work\\\\\\">\\\\r\\\\n                                    <button class=\\\\\\"change-list__minus change-list__btn icon-minus icon\\\\\\"></button>\\\\r\\\\n                                    <span  class=\\\\\\"change-list__input\\\\\\">25</span>\\\\r\\\\n                                    <button class=\\\\\\"change-list__plus change-list__btn icon-plus icon\\\\\\"></button>\\\\r\\\\n                                </div>\\\\r\\\\n                            </div>\\\\r\\\\n                            <p class=\\\\\\"change-list__details\\\\\\">\\\\r\\\\n                                Please select a value between 15 and 25 <span class=\\\\\\"change-list__details--white\\\\\\">minutes</span>\\\\r\\\\n                            </p>\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"change-list__item change-list__item--other\\\\\\">\\\\r\\\\n                            <div class=\\\\\\"change-list__info\\\\\\">\\\\r\\\\n                                <h4 class=\\\\\\"change-list__title\\\\\\">Work Iteration</h4>\\\\r\\\\n                                <div class=\\\\\\"change-list__setting\\\\\\" id=\\\\\\"iteration\\\\\\">\\\\r\\\\n                                    <button class=\\\\\\"change-list__minus change-list__btn icon-minus icon\\\\\\"></button>\\\\r\\\\n                                    <span  class=\\\\\\"change-list__input\\\\\\">5</span>\\\\r\\\\n                                    <button class=\\\\\\"change-list__plus change-list__btn icon-plus icon\\\\\\"></button>\\\\r\\\\n                                </div>\\\\r\\\\n                            </div>\\\\r\\\\n                            <p class=\\\\\\"change-list__details\\\\\\">\\\\r\\\\n                                Please select a value between 2 and 5 <span class=\\\\\\"change-list__details--white\\\\\\">iterations</span>\\\\r\\\\n                            </p>\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"change-list__item change-list__item--education\\\\\\">\\\\r\\\\n                            <div class=\\\\\\"change-list__info\\\\\\">\\\\r\\\\n                                <h4 class=\\\\\\"change-list__title\\\\\\">Short Break</h4>\\\\r\\\\n                                <div class=\\\\\\"change-list__setting\\\\\\" id=\\\\\\"shortBreak\\\\\\">\\\\r\\\\n                                    <button class=\\\\\\"change-list__minus change-list__btn icon-minus icon\\\\\\"></button>\\\\r\\\\n                                    <span  class=\\\\\\"change-list__input\\\\\\">5</span>\\\\r\\\\n                                    <button class=\\\\\\"change-list__plus change-list__btn icon-plus icon\\\\\\"></button>\\\\r\\\\n                                </div>\\\\r\\\\n                            </div>\\\\r\\\\n                            <p class=\\\\\\"change-list__details\\\\\\">\\\\r\\\\n                                Please select a value between 3 and 5 <span class=\\\\\\"change-list__details--white\\\\\\">minutes</span>\\\\r\\\\n                            </p>\\\\r\\\\n                        </li>\\\\r\\\\n                        <li class=\\\\\\"change-list__item change-list__item--hobby\\\\\\">\\\\r\\\\n                            <div class=\\\\\\"change-list__info\\\\\\">\\\\r\\\\n                                <h4 class=\\\\\\"change-list__title\\\\\\">Long Break</h4>\\\\r\\\\n                                <div class=\\\\\\"change-list__setting\\\\\\" id=\\\\\\"longBreak\\\\\\">\\\\r\\\\n                                    <button class=\\\\\\"change-list__minus change-list__btn icon-minus icon\\\\\\"></button>\\\\r\\\\n                                    <span  class=\\\\\\"change-list__input\\\\\\">15</span>\\\\r\\\\n                                    <button class=\\\\\\"change-list__plus change-list__btn icon-plus icon\\\\\\"></button>\\\\r\\\\n                                </div>\\\\r\\\\n                            </div>\\\\r\\\\n                            <p class=\\\\\\"change-list__details\\\\\\">\\\\r\\\\n                                Please select a value between 15 and 30 <span class=\\\\\\"change-list__details--white\\\\\\">minutes</span>\\\\r\\\\n                            </p>\\\\r\\\\n                        </li>\\\\r\\\\n                    </ul>\\\\r\\\\n                    <section class=\\\\\\"cycle\\\\\\">\\\\r\\\\n                        <h2 class=\\\\\\"settings__title\\\\\\">Your cycle</h2>\\\\r\\\\n                        <div class=\\\\\\"cycle__container\\\\\\">\\\\r\\\\n                        </div>\\\\r\\\\n                    </section>\\\\r\\\\n                    <div class=\\\\\\"settings__buttons service-btn\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"service-btn__btn service-btn__btn--blue settings__buttons-return\\\\\\">Go to Task</button>\\\\r\\\\n                        <button class=\\\\\\"service-btn__btn service-btn__btn--green settings__buttons-save\\\\\\">Save</button>\\\\r\\\\n                    </div>\\\\r\\\\n\\\\r\\\\n                </div>\\\\r\\\\n            </div>\\\\r\\\\n        </section>\\\\r\\\\n\\\\r\\\\n\\\\r\\\\n\\\\r\\\\n    </main>\\""`;
